### ISSUE TITLE
OOM exception using startFixedUpload

### ISSUE THREAD

**Issue by AlexZvonikPrevacio** [akinaru/speed-test-lib#14]
I face OOM exception using startFixedUpload with these parameters. I uses lots of memory. About 100Mb on HTC One api 5.0.2
```
public static final String SPEED_TEST_SERVER_HOST = "2.testdebit.info";
public static final int SPEED_TEST_SERVER_PORT = 80;
public static final String SPEED_TEST_SERVER_URI_DL = "/fichiers/1000Mo.dat";
public static final int FILE_SIZE = 20_000_000 ;
public static final int SPEED_TEST_MAX_DURATION = 10000;
public static final int SPEED_TEST_REPORT_INTERVAL = 500;

speedTestSocket.startFixedUpload(
   SPEED_TEST_SERVER_HOST,
   SPEED_TEST_SERVER_PORT,
   SPEED_TEST_SERVER_URI_DL,
   FILE_SIZE,
   SPEED_TEST_MAX_DURATION,
   SPEED_TEST_REPORT_INTERVAL);
```
```
10-17 17:31:26.681 4041-4094/pervacio.com.signalstrength I/art: Forcing collection of SoftReferences for 95MB allocation
10-17 17:31:26.721 4041-4094/pervacio.com.signalstrength E/art: Throwing OutOfMemoryError "Failed to allocate a 100000012 byte allocation with 8388512 free bytes and 94MB until OOM"
10-17 17:31:26.721 4041-4094/pervacio.com.signalstrength E/AndroidRuntime_2_crash: crash in the same process: Thread-4674
     java.lang.OutOfMemoryError: Failed to allocate a 100000012 byte allocation with 8388512 free bytes and 94MB until OOM
         at java.util.Arrays.copyOfRange(Arrays.java:2671)
         at java.util.Arrays.copyOf(Arrays.java:2455)
         at fr.bmartel.speedtest.RandomGen.nextArray(RandomGen.java:91)
         at fr.bmartel.speedtest.SpeedTestTask.writeUpload(SpeedTestTask.java:231)
         at fr.bmartel.speedtest.SpeedTestSocket.startUpload(SpeedTestSocket.java:546)
         at fr.bmartel.speedtest.SpeedTestSocket.startFixedUpload(SpeedTestSocket.java:360)
         at fr.bmartel.speedtest.SpeedTestSocket.startFixedUpload(SpeedTestSocket.java:383)
         at com.signalstrength.ui.fragment.PlaceholderFragment2.lambda$new$2(PlaceholderFragment2.java:110)
         at com.signalstrength.ui.fragment.PlaceholderFragment2$$Lambda$3.execute(Unknown Source)
         at com.signalstrength.Router.startTask(Router.java:42)
         at com.signalstrength.Router.onSpeedListenerFinish(Router.java:26)
         at com.signalstrength.speedListeners.AbstractSpeedListener.onStop(AbstractSpeedListener.java:98)
         at com.signalstrength.speedListeners.AbstractSpeedListener.onTimerStop(AbstractSpeedListener.java:70)
         at com.signalstrength.StopWaiter.run(StopWaiter.java:34)
```

**Comment by akinaru** [comment]
Hi, thank you for reporting this, I will add an API to configure File storage to store uploaded file in File storage instead of RAM and implement some tests concerning app memory usage

**Comment by AlexZvonikPrevacio** [comment]
Thanks for the quick response. I'll be waiting for your fix.

**Comment by akinaru** [comment]
It is fixed in `1.23`, by default RAM storage is used 
![image](https://cloud.githubusercontent.com/assets/5183022/19476993/63280c22-953c-11e6-95c7-b44490d94c5a.png)
You can use file storage using before uploading (check https://github.com/akinaru/speed-test-lib#set-upload-file-storage-type) :
```
speedTestSocket.setUploadStorageType(UploadStorageType.FILE_STORAGE);
```
![image](https://cloud.githubusercontent.com/assets/5183022/19477054/b2e140ee-953c-11e6-8a72-37518f079c1c.png)
The temporary file is automatically removed when upload is finished

Note that I have refactor a few things in `1.23` so you may have some imports than have been changed

**Comment by AlexZvonikPrevacio** [comment]
Great. Thank you for fast reply.
